### ISSUE TITLE
Collaborative editing of Forms

### ISSUE THREAD

**Issue by Pilzinsel64** [nextcloud/forms#322]
Hello!

It would be really nice if we have collaborative editing of Forms.

### Usage Examples
- The whole marketing team has access to some Forms and responses.
- My situation: I have a team of some people. They review user requests (bug reports and feature request) for my public programs. They talk with the user and create cards on the correct board in the Deck app. But only I have access to the form response and that makes it a bit harder. I have to export all response as excel/csv and send it to them every time new responses are here.

### What do I think about how It could be used
Share a Form to other users/groups and define if they have access to view the responses and if they have access to edit the Form. It should also be possible then to allow/deny re-sharing the Form.
As a good example, take the Deck app.

This feature has been discussed already here: #316

**Comment by jancborchardt** [comment]
I split the part of **Ability to share responses #455** out because that one is reasonable and simpler to do.

Collaborative editing of Forms comes with all the pitfalls of conflict handling etc and it’s not something we should get into yet at this stage. :)

**Comment by ostasevych** [comment]
> I split the part of **Ability to share responses #455** out because that one is reasonable and simpler to do.
> 
> Collaborative editing of Forms comes with all the pitfalls of conflict handling etc and it’s not something we should get into yet at this stage. :)

Is it possible to re-use some how some parts of the Text app, which allows to organise collaboration in creation of very simple forms, eg checkboxes?..

**Comment by jancborchardt** [comment]
@ostasevych I don’t think so, as in Nextcloud Text we use [ProseMirror](https://prosemirror.net/) as base which is focused on text editing. But ref @juliushaertl for info.

**Comment by juliushaertl** [comment]
Yes, I don't think that using prosemirror and its document format here will make a lot of sense. For some collaborative editing, you could of course use a somehow similar change tracking based sync format as the text app does, but this will most likely require a larger amount of work in restructuring the database for that.

**Comment by skjnldsv** [comment]
Yeah, for me it's far too complicated.
We could have a way to allow working on different questions, adding/removing.
But we cannot implement a conflict resolution (the hardes part) easily.

I'd either close as wontfix or just partial one

**Comment by jancborchardt** [comment]
It’s set for 4.0 so not urgent, but also definitely not wontfix. It already came up in our own little usecase that we needed the marketing team and me to have access to the Forms.

**Comment by splashote** [comment]
Great to read that this will be adressed.
Imho it's kind of misleading, that there is the sharing function with other users for forms at all

**Comment by xplosionmind** [comment]
Hello, I ended up here since my and my team heavily rely on Nextcloud and if only one of us can create and edit a form without collaboration by the others, this implementation is almost useless. If I may: is it really necessary to define conflicts management? I feel like if the user just knows forms have to be edited one at a time, this integration could be **very** useful and not that hard to implement.

Thanks a lot for your work and your time!

T.

**Comment by shukon** [comment]
Extending the thoughts of @xplosionmind , a simple mutex-solution would be fully satisfying imho. Just have a lock on the form while it is being edited and a timeout if one forgets to free it afterwards.
Conflict handling is mainly needed for realtime or offline collaborative editing. Offline is not possible anyway (it's browser only, right?) and realtime is really cool but only necessary in corner cases. :)

**Comment by tonyhts** [comment]
I'm waiting this feature for months. I hope that doesn't take long.  My job is on the table. 
Sending good vibes to the dev team.
Thxx

**Comment by bitbacchus** [comment]
An idea to allow for a workaround might be to implement import and export of forms (#582). Users could then share their exported, say CSV, files. The different exported CSV files could be manually merged in a spreadsheet app and imported to the final form.

This would at least work for our use case. What do you think?

**Comment by k33la1n0** [comment]
hi there,

could the survey (forms) be provided with an edit(orange)/public(green) button (i.e. a button that changes color and function by click). when the button is set to edit, the edit option for groups or nextcloud-user or a public-edit-link appears. and when the button is clicked to public, the edit-group/nextcloud-user/public-edit-link is blocked and can only be edited by the creator of the survey (forms)?

or all people added as editors to the survey (forms) can click the edit/public button to 'lock' or 'unlock' the survey?

thx for your work, i like this project <3

**Comment by agentlibre** [comment]
Perhaps **Etherpad** could be useful in some way to meet this challenge: https://etherpad.org/

**Comment by ajoana** [comment]
> Extending the thoughts of @xplosionmind , a simple mutex-solution would be fully satisfying imho. Just have a lock on the form while it is being edited and a timeout if one forgets to free it afterwards. Conflict handling is mainly needed for realtime or offline collaborative editing. Offline is not possible anyway (it's browser only, right?) and realtime is really cool but only necessary in corner cases. :)

I'd be happy to just have a group or circle with the right to edit - only one at a time. I'm just a user but wouldn't a file lock be enough while one is editing?

**Comment by rosa2** [comment]
Hello and thanks
we need it too :)

**Comment by muehlburger** [comment]
This would be really handy!

**Comment by rosa2** [comment]
Hello and thanks for the app,
I came here because we also need this feature. In our case, the communication team needs to edit the forms that other teams make.
Thanks a lot and have a good day :)

**Comment by Thatoo** [comment]
Any update on the development of this feature?

**Comment by RalfJung** [comment]
We don't need collaborative editing, but just having more than one person who has the right to edit a form would be really useful when preparing a form together. It doesn't need to do any conflict resolution, just let me edit text in a form someone else created. That would be nice. :)

**Comment by Chartman123** [comment]
@merlin-tc this is already implemented
